### DISCUSSION TITLE
Album or song is already in a playlist

### DISCUSSION THREAD

**Discussion by Domiziano-R** [navidrome/navidrome#3099] [Ideas]
Hello there, 
could be usefull have the possibility to know if an album or a song is already in a playlist and which playlist

**Comment by racehd** [comment]
This would be a nice enhancement. I picture alongside the song "Get Info" would make a good place for this. There currently is already a "Mapped Tags" and "Raw Tags". a third section "Playlists" could be introduced, or more generically "Additional Info" (extendable for other misc info like this)

Implementation would involve:
- Backend: SQL query to get playlists for a song/album
- API: create or update endpoints (I have not looked into this portion yet). 
- Frontend: New tab in the song/album info dialog

As an example, for implementing for songs I'd create a SQL view: 
```sql
CREATE VIEW song_playlists AS
SELECT 
    mf.id as media_file_id,
    mf.title as song_title,
    mf.artist as song_artist,
    p.id as playlist_id,
    p.name as playlist_name,
    p.owner_id as playlist_owner_id,
    p.public as playlist_public
FROM media_file mf
JOIN playlist_tracks pt ON mf.id = pt.media_file_id
JOIN playlist p ON pt.playlist_id = p.id;
```

and then queries could use the `song.id` and `user.ID` variables to return playlists that belong to that user,  or are public. 
```sql
SELECT playlist_name FROM song_playlists 
WHERE media_file_id = song.id
  AND (playlist_owner_id = user.ID OR playlist_public = 1)
ORDER BY playlist_name;
```

Could rework a bit to include albums as well (I'd include the `song_title` alongside `playlist_name` so you'd see which song(s) are in playlists.

**Comment by deluan** [comment]
@racehd This is already implemented in the latest versions:


https://github.com/user-attachments/assets/bf67abdb-781f-44e1-bd13-45e8429e4c19

**Comment by racehd** [comment]
Thanks, looks like my docker image was a bit more than month old. my apologies!
